wikipedia_cleanup/predictor.py

```diff
@@ -0,0 +1,114 @@
+import random
+from abc import ABC, abstractmethod
+from datetime import date, datetime, timedelta
+from typing import List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+
+
+class Predictor(ABC):
+    @abstractmethod
+    def fit(
+        self, train_data: pd.DataFrame, last_day: datetime, keys: List[str]
+    ) -> None:
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_relevant_attributes() -> List[str]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def predict_timeframe(
+        self,
+        data_key: np.ndarray,
+        additional_data: np.ndarray,
+        columns: List[str],
+        first_day_to_predict: date,
+        timeframe: int,
+    ) -> bool:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_relevant_ids(self, identifier: Tuple) -> List[Tuple]:
+        raise NotImplementedError()
+
+
+class ZeroPredictor(Predictor):
+    def fit(
+        self, train_data: pd.DataFrame, last_day: datetime, keys: List[str]
+    ) -> None:
+        pass
+
+    def predict_timeframe(
+        self,
+        data_key: np.ndarray,
+        additional_data: np.ndarray,
+        columns: List[str],
+        first_day_to_predict: date,
+        timeframe: int,
+    ) -> bool:
+        return False
+
+    def get_relevant_ids(self, identifier: Tuple) -> List[Tuple]:
+        return [identifier]
+
+    @staticmethod
+    def get_relevant_attributes() -> List[str]:
+        return []
+
+
+class OnePredictor(ZeroPredictor):
```

Comment:
The point of inheriting from `ZeroPredictor` is to not implement `.fit`, `.get_relevant_ids` and `.get_relevant_attributes`, right?
If that's the case, I suggest having a `StaticPredictor` class from which `ZeroPredictor` inherits as well to make it explicit.